django/test/testcases.py

```diff
@@ -144,7 +142,7 @@ def __exit__(self, exc_type, exc_value, traceback):
 class _AssertTemplateNotUsedContext(_AssertTemplateUsedContext):
     def test(self):
         self.test_case.assertFalse(
-            self.template_name in self.rendered_template_names,
+            self.template_name in [t.name for t in self.rendered_templates],
```

Comment:
Ignoring unnamed template names in all `assertTemplate.*Used` helpers are more consistent.